addons/hw_drivers/iot_handlers/drivers/KeyboardUSBDriver_L.py

```diff
@@ -299,7 +297,22 @@ def _set_device_type(self, device_type='keyboard'):
                 device.grab()
             self.read_barcode_lock = Lock()
         else:
-            self.device_type = 'keyboard'
+            # Check if the keyboard is a touchscreen
+            udev_output = subprocess.run(
+                ["udevadm", "info", "--export-db"], capture_output=True, text=True, check=True
+            ).stdout.split("\n\n")
+
+            # If the device is a touchscreen, we set the device type to 'touchscreen', and rename the device
+            self.device_type, self.device_name = next(
```

Comment:
Here you are specifying the `device_type` as "touchscreen".
Then this value is used in `send_all_devices` when calling the route `/iot/setup`.
The check `if data_device['type'] in available_types` implies there is a "touchscreen" type in the iot_device model's `type` field which isn't the case, so the device is never being created as an entry in the database